File: packages/templates/clients/websocket/python/example-slack.py
import sys
import os
import time

sys.path.append(os.path.join(os.path.dirname(__file__), 'test'))

from temp.snapshotTestResult.client_slack.client import SlackWebsocketAPIClientClient

def custom_message_handler(message):
    print(f"\033[94mIncomming event from Slack\033[0m: {message}")

def main():
    client = SlackWebsocketAPIClientClient()
    client.register_message_handler(custom_message_handler)

    client.connect()

    # Keep program alive for a while to allow message processing
    time.sleep(1000)

main()

File: packages/templates/clients/websocket/javascript/example-slack.js
const WSClient = require('./test/temp/snapshotTestResult/client_slack/client.js');
// Example usage
const wsClient = new WSClient();

// Example of how custom message handler that operates on incoming messages can look like
function myHandler(message) {
  console.log('====================');
  console.log('\x1b[94mIncoming event from Slack\x1b[0m:', message);
  console.log('====================');
}

async function main() {
  wsClient.registerMessageHandler(myHandler);

  try {
    await wsClient.connect();

    // Keep the process alive to receive Slack events
    // The connection will stay open until you terminate the process
    console.log('Listening for Slack events... Press Ctrl+C to exit.');
    await new Promise(() => {});
  } catch (error) {
    console.error('Failed to connect to WebSocket:', error.message);
  }
}

main();
